hydromt/vector.py

```diff
@@ -740,6 +740,10 @@ def from_gdf(
         if index_dim is None:
             index_dim = gdf.index.name if gdf.index.name is not None else "index"
         geom_name = gdf.geometry.name
+        # Some index dimension checking
+        # TODO Check this!!! Might be unwanted
+        if index_dim in gdf.columns:
+            gdf.set_index(index_dim, inplace=True)
```

Comment:
I like to avoid changing the input data as the use may want to continue using gdf outside of this function. Therefore better to not use the inplace=True argument.

```suggestion
            gdf = gdf.set_index(index_dim)
```

Reply:
```python
gdf.set_index(index_dim, inplace=True)
```
I was kinda wondering the same thing. We should, I think, discuss this, because there is a trade-off in terms of memory. Most of the time you wont get near memory issues, but it is something to keep in mind when copying the thing.
So I agree but we could discuss it.

Reply:
Changed it.